test/Mono.Linker.Tests.Cases/Warnings/WarningSuppression/DetectRedundantSuppressionsInMembersAndTypesUsingTarget.cs

```diff
@@ -0,0 +1,89 @@
+﻿// Copyright (c) .NET Foundation and contributors. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Diagnostics.CodeAnalysis;
+using System.Linq.Expressions;
+using Mono.Linker.Tests.Cases.Expectations.Assertions;
+using Mono.Linker.Tests.Cases.Expectations.Metadata;
+
+[module: UnconditionalSuppressMessage ("Test", "IL2071", Scope = "type", Target = "T:Mono.Linker.Tests.Cases.Warnings.WarningSuppression.DetectRedundantSuppressionsInMembersAndTypesUsingTarget.RedundantSuppressionOnType")]
+[module: UnconditionalSuppressMessage ("Test", "IL2071", Scope = "member", Target = "M:Mono.Linker.Tests.Cases.Warnings.WarningSuppression.DetectRedundantSuppressionsInMembersAndTypesUsingTarget.RedundantSuppressionOnMethod.Test")]
+[module: UnconditionalSuppressMessage ("Test", "IL2071", Scope = "type", Target = "T:Mono.Linker.Tests.Cases.Warnings.WarningSuppression.DetectRedundantSuppressionsInMembersAndTypesUsingTarget.RedundantSuppressionOnNestedType.NestedType")]
+[module: UnconditionalSuppressMessage ("Test", "IL2071", Scope = "member", Target = "M:Mono.Linker.Tests.Cases.Warnings.WarningSuppression.DetectRedundantSuppressionsInMembersAndTypesUsingTarget.RedundantSuppressionOnProperty.get_TrimmerCompatibleProperty")]
+
+namespace Mono.Linker.Tests.Cases.Warnings.WarningSuppression
+{
+	[ExpectedNoWarnings]
+	[SkipKeptItemsValidation]
+	public class DetectRedundantSuppressionsInMembersAndTypesUsingTarget
+	{
+		public static void Main ()
+		{
+			RedundantSuppressionOnType.Test ();
+			RedundantSuppressionOnMethod.Test ();
+			RedundantSuppressionOnNestedType.Test ();
+			RedundantSuppressionOnProperty.Test ();
+		}
+
+		public static Type TriggerUnrecognizedPattern ()
+		{
+			return typeof (DetectRedundantSuppressionsInMembersAndTypesUsingTarget);
+		}
+
+		public static string TrimmerCompatibleMethod ()
+		{
+			return "test";
+		}
+
+		[ExpectedWarning ("IL2121", "IL2071", ProducedBy = ProducedBy.Trimmer)]
+		public class RedundantSuppressionOnType
+		{
+			public static void Test ()
+			{
+				DetectRedundantSuppressionsInMembersAndTypesUsingTarget.TrimmerCompatibleMethod ();
+			}
+		}
+
+		public class RedundantSuppressionOnMethod
+		{
+			[ExpectedWarning ("IL2121", "IL2071", ProducedBy = ProducedBy.Trimmer)]
+			public static void Test ()
+			{
+				DetectRedundantSuppressionsInMembersAndTypesUsingTarget.TrimmerCompatibleMethod ();
+			}
+		}
+
+		public class RedundantSuppressionOnNestedType
+		{
+			public static void Test ()
+			{
+				NestedType.TrimmerCompatibleMethod ();
+			}
+
+			[ExpectedWarning ("IL2121", "IL2071", ProducedBy = ProducedBy.Trimmer)]
```

Comment:
We could also create an issue for this (just like the XML ones) - the warning should be reported on the assembly in this case (since the suppression is on the assembly). But it's a low priority.
In either case I would at the very least add a comment about this discrepancy.

Reply:
For now I added a comment stating the discrepancy.